airbyte-cdk/bulk/core/base/src/testFixtures/kotlin/io/airbyte/cdk/output/BufferingOutputConsumer.kt

```diff
@@ -79,4 +80,11 @@ class BufferingOutputConsumer(
     fun traces(): List<AirbyteTraceMessage> = synchronized(this) { listOf(*traces.toTypedArray()) }
 
     fun messages(): List<AirbyteMessage> = synchronized(this) { listOf(*messages.toTypedArray()) }
+
```

Comment:
optional, but might be worth including in this PR:

```suggestion

    fun resetNewMessagesCursor() { 
        synchronized(this) { messagesIndex = 0 }
    }

```

or, alternatively, a `reset()` method that just nukes everything, not just the cursor